### PR TITLE
MM-39634: Migrated gorp to sqlx

### PR DESCRIPTION
**Pull request by mr-aboutin**
#### Summary

Migration from gorp to sqlx for `terms_of_service_store.go`

#### Ticket Link

Fixes: https://github.com/mattermost/mattermost-server/issues/18872

#### Release Note

```release-note
Migration from gorp to sqlx in `terms_of_service_store.go
Removed redundant check for Get to check if TermsOfService is found since an error will be returned if no row was found
```

### CODE REVIEW

File: store/sqlstore/terms_of_service_store.go [hunk 1]
```diff
@@ -66,23 +69,31 @@ func (s SqlTermsOfServiceStore) GetLatest(allowFromCache bool) (*model.TermsOfSe
...
+	var termsOfService model.TermsOfService
+	queryString, _, err := s.getQueryBuilder().
+		Select("*").
+		From("TermsOfService").
+		Where("id = ?").
+		ToSql()
+
 	if err != nil {
-		return nil, errors.Wrapf(err, "could not find TermsOfService with id=%s", id)
+		return nil, err
```
Comment (agnivade) [inline store/sqlstore/terms_of_service_store.go:91]: This needs to be `return nil, errors.Wrap(err, "terms_of_service_to_sql")` to maintain consistency with similar errors like this.

File: store/sqlstore/terms_of_service_store.go [hunk 2]
```diff
@@ -66,23 +69,31 @@ func (s SqlTermsOfServiceStore) GetLatest(allowFromCache bool) (*model.TermsOfSe
...
 	}
-	if obj == nil {
-		return nil, store.NewErrNotFound("TermsOfService", id)
+
+	err = s.GetReplicaX().Get(&termsOfService, queryString, id)
+	if err != nil {
+		return nil, errors.Wrapf(err, "could not find TermsOfService with id=%s", id)
 	}
```
Comment (agnivade) [inline store/sqlstore/terms_of_service_store.go:97]: This needs to be similar to what we are doing in L72.

```go
if err != nil {
  if err == sql.ErrNoRows {
     return nil, store.NewErrNotFound("TermsOfService", "id")
  }
  return nil, errors.Wrapf(err, "could not find TermsOfService with id=%s", id)
}
```

**Review by agnivade — COMMENTED** [review]
Great stuff @mr-aboutin ! Just some small things to fix. Also please run a `make golangci-lint` to fix the goimports errors.

**Review by agnivade — APPROVED** [review]
Looks great!

**Review by isacikgoz — APPROVED** [review]
LGTM, thank you @mr-aboutin 👍

**Pull request merged**